### PR TITLE
Support GHC <= 9.2

### PR DESCRIPTION
**Pull request by jonathanlking**
* Drop Nix based CI pipeline and switch to [haskell/actions](https://github.com/haskell/actions) (as described in https://kodimensional.dev/github-actions).
* Add GHC 8.10.8, 9.0.2 and 9.2.2 (Cabal only, as there is not Stack LTS yet) to the test matrix.
* Add support for GHC 8.10.7, 9.0.2 and 9.2.2 while maintaining compatibility for 8.6.5 and 8.8.4 (using CPP pragmas)

### CODE REVIEW

File: src/Options/Harg/Single.hs [hunk 1]
```diff
@@ -59,4 +59,4 @@ instance B.TraversableB (Single a) where
 
 instance B.ApplicativeB (Single a) where
   bprod (Single l) (Single r) = Single (P.Pair l r)
-  bpure = Single
+  bpure x = Single x
```
Comment (jonathanlking) [inline src/Options/Harg/Single.hs:62]: I believe this is required due to the new [simplified subsumption](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0287-simplify-subsumption.rst) (in GHC > 9.0)

File: harg.cabal [hunk 2]
```diff
@@ -67,9 +67,9 @@ library
     ghc-options:        -Wall -Wno-unticked-promoted-constructors
     build-depends:
         base >=4.7 && <5,
-        aeson >=1.4.2 && <1.5,
+        aeson >=2.0 && <2.1,
```
Comment (jonathanlking) [inline harg.cabal:70]: Stackage LTS's that support GHC 9 also require `aeson` 2.
I'm unfamiliar with how to support multiple library versions, but I _think_ it's possible with CPP macros to support both versions of `aeson` (if necessary).

File: src/Options/Harg/Het/Prod.hs [hunk 3]
```diff
@@ -15,12 +15,13 @@ where
 import qualified Barbies as B
 import Data.Aeson ((.!=), (.:?))
 import qualified Data.Aeson as JSON
+import qualified Data.Aeson.Key as JSON.Key
```
Comment (jonathanlking) [inline src/Options/Harg/Het/Prod.hs:18]: This is new from `aeson` 2

File: src/Options/Harg/Nested.hs [hunk 4]
```diff
@@ -23,15 +23,6 @@ import qualified Data.Generic.HKD as HKD
...
 import GHC.Generics (Generic)
 
--- Orphan HKD FromJSON instance
-instance
-  JSON.GFromJSON JSON.Zero (HKD.HKD_ f structure) =>
-  JSON.FromJSON (HKD.HKD structure f)
-  where
-  parseJSON =
-    fmap HKD.HKD
-      . JSON.gParseJSON JSON.defaultOptions JSON.NoFromArgs
```
[the remark this answers is not in the document]
Comment (alexpeits) [reply src/Options/Harg/Nested.hs:33]: I think we can use generic-aeson here?

**Review by alexpeits — DISMISSED** [review]
Approving, I will create an issue to revisit the `FromJSON` instance for `HKD`

**Review by alexpeits — APPROVED** [review]

**Pull request merged**